fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1346,6 +1352,242 @@ def assert_state(self, state: Union[TrainingState, List[TrainingState]]) -> None
                 traceback.print_stack()
             raise ValueError(msg)
 
+    # Optim State dict functions
+
+    def consolidate_optim_state_dict(self, optim: torch.optim.Optimizer, recipient_rank: int = 0) -> None:
+        """Update the consolidated state_dict list, one per rank.
+
+        Arguments:
+            recipient_rank (int): on which rank to materialize the full state dict.
+            -1 is a special value, which means that all ranks should have the state
+
+        .. warning: This needs to be called on all replicas"""
+        _default_device = torch.device("cuda")
```

Comment:
how about calling `self._lazy_init()` and then using `self.compute_device`? That's what we're doing for `apply`/`summon_all_params`

It also makes it work in case the user hasn't set the current_device properly